### ISSUE TITLE
Unable to enable logging on the lldb-server host side.

### ISSUE THREAD

**Issue by tzb99** [llvm/llvm-project#56665]
The lldb-server command is:
```shell
lldb-server g 0.0.0.0:2345 riscvv --log-channels lldb all
```

I cross compiled the lldb project using a in-tree build and the source patch is mainly from: https://github.com/luismarques/llvm-project/commit/4a0cccb0cfa8cb5c59c8803077a76751498447ec

```console
1658443389.905858755 ProcessLaunchInfo.cpp:SetUpPtyRedirection                    Generating a pty to use for stdin/out/err
1658443393.688252211 GDBRemoteCommunicationServerLLGS::InitializeDelegate called with NativeProcessProtocol pid 133, current state: invalid
1658443393.697373867 NativeThreadLinux::SetStoppedBySignal called with signal 0x13
1658443393.700402975 NativeThreadLinux.cpp:MaybeLogStateChange                    pid=133, tid=133: changing from state invalid to stopped
1658443393.705924034 GDBRemoteCommunicationServerLLGS.cpp:LaunchProcess           pid = 133: setting up stdout/stderr redirection via $O gdb-remote commands
1658443393.707329988 ProcessGDBRemoteCommunicationServerLLGS::LaunchProcess setting inferior STDIO fd to 3
Launched 'riscvv' as process 133...
lldb-server-local_build
Connection established.
1658443432.083425999 GDBRemoteCommunicationServerLLGS.cpp:SetCurrentThreadID      setting current thread id to 133
1658443432.084943056 GDBRemoteCommunicationServerLLGS.cpp:PrepareStopReplyPacketF preparing packet for pid 133 tid 133
1658443432.086048365 LogThreadStopInfo: m_stop_info in thread: signal 0x13
1658443432.086704016 LogThreadStopInfo: returned stop_info: signal 0x13
1658443432.087290525 GDBRemoteCommunicationServerLLGS.cpp:PrepareStopReplyPacketF pid 133, tid 133, got signal signo = 19, reason = 5, exc_type = 0
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: lldb-server g 0.0.0.0:2345 riscvv --log-channels lldb all
1.	Program arguments: lldb-server g 0.0.0.0:2345 riscvv --log-channels lldb all
#0 0x00000000000fc97e PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
[ 5430.048226] lldb-server[132]: unhandled signal 11 code 0x1 at 0x0000000000000000 in lldb-server[10000+f9d000]
[ 5430.050003] CPU: 0 PID: 132 Comm: lldb-server Not tainted 5.18.0-rc2 #1
[ 5430.050525] Hardware name: riscv-virtio,qemu (DT)
[ 5430.050923] epc : 000000000016e592 ra : 000000000016e588 sp : 00ffffffc5ced4e0
[ 5430.051341]  gp : 0000000001051ed0 tp : 00ffffffb894f7b0 t0 : fffffffffffff9e0
[ 5430.051815]  t1 : 000000000007275c t2 : 0000000000000010 s0 : 00ffffffc5ced868
[ 5430.052229]  s1 : 00000000010a5110 a0 : 0000000000daf1c8 a1 : 00ffffffc5ced475
[ 5430.052682]  a2 : 000000000000004c a3 : 0000000000000001 a4 : 00ffffffc5ced468
[ 5430.053065]  a5 : 00ffffffc5ced450 a6 : 00000000001476aa a7 : 0000000000000000
[ 5430.053688]  s2 : 00000000010d63e8 s3 : 00ffffffc5ced600 s4 : 0000000000000000
[ 5430.054169]  s5 : 00000000010ed420 s6 : 00000000010edf40 s7 : 00ffffffc5ced530
[ 5430.054577]  s8 : 0000000000000000 s9 : 00ffffffc5ced5a8 s10: 00ffffffc5ced6b0
[ 5430.054959]  s11: 00ffffffc5ced688 t3 : 00ffffffb89c70b6 t4 : 0000000000000000
[ 5430.055240]  t5 : 00ffffffc5cec978 t6 : 000000000000002a
[ 5430.055438] status: 0000000200004020 badaddr: 0000000000000000 cause: 000000000000000d
Segmentation fault
```

**Comment by DavidSpickett** [comment]
If I understand correctly, you're using source that isn't the current `main`. So I'm hesitant to look too much into this unless it's reproducible (or at least theoretically reproducible) elsewhere.

(I'm adding the riscv label for now we can remove it if it is a generic issue)

I don't know if you're aware, there is another effort to get riscv lldb-server going - https://reviews.llvm.org/D128250 or perhaps they are one and the same. If this issue happens with those changes it could be valuable feedback for them.

If you want to dig a bit more, the crash point appears to be in `lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationServerLLGS.cpp` `GDBRemoteCommunicationServerLLGS::PrepareStopReplyPacketForThread`. On the surface it looks like the logging itself is crashing but it could be misleading. A debug build of lldb-server, run under, well, gdb I guess :) would help.

**Comment by DavidSpickett** [comment]
Sorry I misread, you're targeting riscv 32 an the patch I linked is dealing with riscv64. Anyway, see if it's related.

**Comment by tzb99** [comment]
> Sorry I misread, you're targeting riscv 32 an the patch I linked is dealing with riscv64. Anyway, see if it's related.

Hi:

Thank you so much for the quick reply! I checked my code and indeed it inherited from https://reviews.llvm.org/D128250, but I will still run the in-tree build again with the debug mode. I ran the debug mode before and if I remembered correctly, the lldb part fails due to the fail of linking libraries. I will report issues here I found new ones. 

Thank you!
